client/my-sites/stats/utils.js

```diff
@@ -32,3 +34,22 @@ export function getPathWithUpdatedQueryString( query = {}, path = page.current )
 
 	return `${ pathname }?${ updatedSearchString }`;
 }
+
+/**
+ * Add analytics event.
+ * @param {*} eventName Analytics event name, automatically prefixed with 'jetpack_odyssey' or 'calypso'
+ * @param {*} properties Analytics properties
+ * @param {*} redirectURL If passed, redirect to the URL after 250ms
+ */
+export const trackStatsAnalyticsEvent = ( eventName, properties, redirectURL = null ) => {
+	const isOdysseyStats = config.isEnabled( 'is_running_in_jetpack_site' );
+
+	// publish an event
+	const event_from = isOdysseyStats ? 'jetpack_odyssey' : 'calypso';
+	recordTracksEvent( `${ event_from }_${ eventName }`, properties );
+
+	// redirect to the Purchase page
+	if ( redirectURL ) {
```

Comment:
It seems this is not currently in use. I wonder if we could remove it. For upgrades, tracking and redirection usually happen together, but I doubt it makes a lot of sense to have them in the same function 🤔

Reply:
That's a good point. I will remove it, and if we need it in the future, we can figure out a way to do it.